arviz/stats/stats.py

```diff
@@ -1299,14 +1294,29 @@ def summary(
                 )
                 extra_metric_names.append(stat_func.__name__)
 
+    metrics: List[xr.Dataset] = []
+    metric_names: List[str] = []
     if extend and kind in ["all", "stats"]:
-        mean = dataset.mean(dim=("chain", "draw"), skipna=skipna)
+        if stat_focus == "mean":
+            mean = dataset.mean(dim=("chain", "draw"), skipna=skipna)
+
+            sd = dataset.std(dim=("chain", "draw"), ddof=1, skipna=skipna)
 
-        sd = dataset.std(dim=("chain", "draw"), ddof=1, skipna=skipna)
+            hdi_post = hdi(dataset, hdi_prob=hdi_prob, multimodal=False, skipna=skipna)
+            hdi_lower = hdi_post.sel(hdi="lower", drop=True)
+            hdi_higher = hdi_post.sel(hdi="higher", drop=True)
+            metrics.extend((mean, sd, hdi_lower, hdi_higher))
+            metric_names.extend(("mean", "sd", f"hdi_{100 * alpha / 2:g}%", f"hdi_{100 * (1 - alpha / 2):g}%"))
+        elif stat_focus == "median":
+            median = dataset.median(dim=("chain", "draw"), skipna=skipna)
 
-        hdi_post = hdi(dataset, hdi_prob=hdi_prob, multimodal=False, skipna=skipna)
-        hdi_lower = hdi_post.sel(hdi="lower", drop=True)
-        hdi_higher = hdi_post.sel(hdi="higher", drop=True)
+            mad = dataset.mad(dim=("chain", "draw"), ddof=1, skipna=skipna)  # not sure if this exists
```

Comment:
I checked the functions in this block and found that all of them are working except the `.mad`. I am not sure about where should this function be added, can you please help.

Reply:
Let's use the one in xarray-einstats that will work directly on xarray datasets like `hdi`.